components/brave_perf_predictor/browser/p3a_bandwidth_savings_permanent_state.h

```diff
@@ -0,0 +1,60 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_PERF_PREDICTOR_BROWSER_P3A_BANDWIDTH_SAVINGS_PERMANENT_STATE_H_
+#define BRAVE_COMPONENTS_BRAVE_PERF_PREDICTOR_BROWSER_P3A_BANDWIDTH_SAVINGS_PERMANENT_STATE_H_
+
+#include <list>
+
+#include "base/macros.h"
+#include "base/timer/timer.h"
+
+class PrefService;
+
+namespace brave_perf_predictor {
+
+// Note: append-only enumeration! Never remove any existing values, as this enum
+// is used to bucket a UMA histogram, and removing values breaks that.
+constexpr std::array<uint64_t, 7> BandwidthSavingsBuckets{
```

Comment:
is there a reason this isn't an enum? If this is a standard way of doing things for histograms that's ok, I'm just not sure

Reply:
since this histogram is based on numeric intervals seemed more straightforward to iterate, but can equally use a series of if-else or a switch statement